pkgs/applications/networking/instant-messengers/teams/default.nix

```diff
@@ -0,0 +1,75 @@
+{ stdenv, fetchurl, makeDesktopItem, wrapGAppsHook
+, alsaLib, atk, at-spi2-atk, at-spi2-core, cairo, cups, dbus, expat, fontconfig, freetype, libgnome-keyring
+, gdk-pixbuf, glib, gtk3, libnotify, libX11, libXcomposite, libXcursor, libXdamage, libuuid, libsecret, libxkbfile
+, libXext, libXfixes, libXi, libXrandr, libXrender, libXtst, nspr, nss, libxcb
+, pango, systemd, libXScrnSaver, libcxx, pulseaudio, dpkg, tree, autoPatchelfHook }:
+
+let
+  binaryName = "teams";
+in stdenv.mkDerivation rec {
+  pname = "teams";
+  version = "1.2.00.32451";
+
+  src = fetchurl {
+    url = "https://packages.microsoft.com/repos/ms-teams/pool/main/t/teams/teams_${version}_amd64.deb";
+    sha256 = "1p053kg5qksr78v2h7cxia5mb9kzgfwm6n99x579vfx48kka1n18";
+  };
+
+  nativeBuildInputs = [ dpkg wrapGAppsHook autoPatchelfHook ];
+
+  unpackPhase = ''
+    dpkg -x $src .
+  '';
+
+  dontWrapGApps = true;
+
+  buildInputs = [
+    libcxx systemd pulseaudio libsecret libxkbfile libgnome-keyring
+    stdenv.cc.cc alsaLib atk at-spi2-atk at-spi2-core cairo cups dbus expat fontconfig freetype
+    gdk-pixbuf glib gtk3 libnotify libX11 libXcomposite libuuid
+    libXcursor libXdamage libXext libXfixes libXi libXrandr libXrender
+    libXtst nspr nss libxcb pango systemd libXScrnSaver
+   ];
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r usr/bin $out/bin
+    cp -r usr/share $out/share
+
+    #mkdir -p $out/{bin,opt/${binaryName},share/pixmaps}
+    #mv * $out/opt/${binaryName}
+
+    #chmod +x $out/opt/${binaryName}/${binaryName}
+    #patchelf --set-interpreter ${stdenv.cc.bintools.dynamicLinker} \
+    #    $out/opt/${binaryName}/${binaryName}
+
+    wrapProgram $out/share/${binaryName}/${binaryName} \
+        "''${gappsWrapperArgs[@]}" \
+        --prefix XDG_DATA_DIRS : "${gtk3}/share/gsettings-schemas/${gtk3.name}/" \
+
```

Comment:
if gtk3, glib, and wrapGAppsHook are present in the proper *inputs, this is completely unnecessary.
`wrapGAppsHook` will wrap automatically, and `glib` setup-hook will collect gtk3's gsettings-schemas directed passed to wrapGAppsHook.